### ISSUE TITLE
Interop with System.Numerics.Tensors

### ISSUE THREAD

**Issue by vpenades** [SciSharp/TensorFlow.NET#534]
A while ago microsoft released System.Numerics.Tensors which includes DenseTensor and SparseTensor types. 

Wouldn't be convenient for Tensorflow.NET to use these types, or at least have an easy way to convert between these types and its own types?

**Comment by Oceania2018** [comment]
@vpenades Can you take a research if possible?

**Comment by vpenades** [comment]
yes, I've using System.Numercs.Tensors myself for a while... The nuget package is here: https://www.nuget.org/packages/System.Numerics.Tensors

Buy I've just realized it's still in preview, I thought it would be more advanced by now, so I believe its's better to wait until the library is finally released

In general I preffer libraries to use BCL types whenever possible. I understand one of the objectives of SciSharp is to align with python, but supporting other types to align with other libraries can be useful too.

I believe the uses cases would be:

- Conversions between SciSharp tensors and numerics tensors.
- Running infereces with numerics tensors

**Comment by Oceania2018** [comment]
@vpenades Anyupdates to refactor NumSharp in `System.Numercs.Tensors` ?

**Comment by vpenades** [comment]
If you see the package release history, there's version 1, and then lots of previews of version 5, which is quite a big jump. I suspect there's lots of changes between 1.0.0 and 5-preview ... so there's no point in doing the integration of version 1.0.0.

And version 5.x.x-preview is still an early release, so I don't know if it's safe to do the implementation.

Anyway, the main idea is this:  There's more and more libraries using machine learning, and the way to comunicate with these libraries is with "tensors"... so it makes sense to be able to do it with a BCL type.

So the plan would be to have in built features to convert from scisharp tensors to system.numerics.tensors and back.

System.Numerics.Tensors is able to wrap an existing System.Buffers.Memory<T> object, so it's able to wrap a pre-existing float[] array without requiring a buffer copy... so if scisharp tensor uses a plan float[] for data storate, it could wrap it efficiently. But I still have a lot to reseach.
